src/server.js

```diff
@@ -209,10 +223,38 @@ function reverseDnsLookup(socket, client) {
 	});
 }
 
+function localAuth(client, user, password, callback) {
+	var result = false;
+	try {
+		result = bcrypt.compareSync(password || "", client.config.password);
+	} catch (error) {
+		if (error === "Not a valid BCrypt hash.") {
+			console.error("User (" + user + ") with no local password set tried signed in. (Probably a ldap user)");
+		}
+		result = false;
+	} finally {
+		callback(result);
+	}
+}
+
+function ldapAuth(client, user, password, callback) {
+	var bindDN = Helper.config.ldap.primaryKey + "=" + user + "," + Helper.config.ldap.baseDN;
```

Comment:
I don't speak LDAP, but wouldn't the user here need to be escaped, for example if `user` here is `Max-P,dn=evil.max-p.me\0`?